src/utils/GcTcpHub.cxx

```diff
@@ -38,17 +38,30 @@
 #include "nmranet_config.h"
 #include "utils/GridConnectHub.hxx"
 
-void GcTcpHub::OnNewConnection(int fd)
+void GcTcpHub::on_new_connection(int fd)
 {
     const bool use_select =
         (config_gridconnect_tcp_use_select() == CONSTANT_TRUE);
-    create_gc_port_for_can_hub(canHub_, fd, nullptr, use_select);
+    {
+        AtomicHolder h(this);
+        numClients_++;
```

Comment:
Not sure if modern compilers are smart enough to substitute a pre-increment if they can, but a pre-increment should be less instruction cycles.

Reply:
The recommendation to use pre-increment instead of post-increment comes from Scott Meyers'  Effective C++ book, and specifically applies to class types that define the pre- and post-increment operator. The most typical place where we encounter such classes are STL iterators. The recommendation was true until about GCC 2.95, beyond which however GCC figured out how to optimize away the copies even from iterators if the implementation available for inlining. It also means that this recommendation has never been true for primitive types.